### PR TITLE
Performance improvements to hbAVSS-light

### PR DESCRIPTION
**Pull request by amiller**
This branch contains several performance improvements for hbAVSS-light
- Uses ThreadPoolExecutor and thread-unsafe operations for scalar multiplication in G1
- Does not verify each merkle branch in ReliableBroadcast, optimistically
- Polynomial evaluation is faster w/ inline horner evaluation

There are also fixes to the dockerfile
- Builds the local directory with `pip install --no-cache -e .[$BUILD]` so it can pick up dev changes
- Builds pairing library without `-e` so it is in release mode (faster)

### CODE REVIEW

File: honeybadgermpc/secretshare_hbavsslight.py [hunk 1]
```diff
@@ -92,6 +103,21 @@ def f(poly, x):
     return y
 
 
+def f_horner(poly, x):
+    assert type(poly) is list
+    # previousb = 0
+    k = len(poly) - 1
+    b = ZR(0)
+    # for (i,coeff) in enumerate(poly):
```
Comment (smkuls) [inline honeybadgermpc/secretshare_hbavsslight.py]: nit: remove commented code?

File: scripts/run_hbavss.py [hunk 2]
```diff
@@ -38,4 +39,6 @@ def main():
 
 if __name__ == "__main__":
     debug = True
+    # cProfile.run("main()")
+    cProfile
```
Comment (smkuls) [inline scripts/run_hbavss.py:43]: That's it? You just write the name of the module and it works? oO

Comment (amiller) [reply scripts/run_hbavss.py:43]: mmm okay really that is playing the role of `# noqa: E{module imported but not used}`

Comment (smkuls) [reply scripts/run_hbavss.py:43]: Why do you need to import it if you aren't using it? Can we not completely remove it?

Comment (amiller) [reply scripts/run_hbavss.py:43]: it's just a script, it's useful to turn on and off that profile output

**Pull request merged**